packages/testnet-faucets/src/ts/handler.ts

```diff
@@ -1,50 +1,92 @@
 
+import { configs } from './configs';
 import { EtherRequestQueue } from './ether_request_queue';
 import { idManagement } from './id_management';
 import { RequestQueue } from './request_queue';
 import { rpcUrls } from './rpc_urls';
 import { utils } from './utils';
 import { ZRXRequestQueue } from './zrx_request_queue';
 
-// HACK: web3 leaks XMLHttpRequest into the global scope and causes requests to hang
-// because they are using the wrong XHR package.
-// Filed issue: https://github.com/ethereum/web3.js/issues/844
-// tslint:disable-next-line:ordered-imports
-import * as Web3 from 'web3';
-
-interface RequestQueueByNetworkId {
-    [networkId: string]: RequestQueue;
+interface ItemByNetworkId<T> {
+    [networkId: string]: T;
 }
 
-enum QueueType {
+enum RequestedAssetType {
     ETH = 'ETH',
+    WETH = 'WETH',
     ZRX = 'ZRX',
 }
 
-const DEFAULT_NETWORK_ID = 42; // kovan
+const FIVE_DAYS_IN_MS = 4.32e8; // TODO: make this configurable
 
 export class Handler {
-    private _etherRequestQueueByNetworkId: RequestQueueByNetworkId = {};
-    private _zrxRequestQueueByNetworkId: RequestQueueByNetworkId = {};
+    private _zeroExByNetworkId: ItemByNetworkId<ZeroEx> = {};
+    private _etherRequestQueueByNetworkId: ItemByNetworkId<RequestQueue> = {};
+    private _zrxRequestQueueByNetworkId: ItemByNetworkId<RequestQueue> = {};
+    private static _dispenseAsset(
+        req: express.Request,
+        res: express.Response,
+        requestQueueByNetworkId: ItemByNetworkId<RequestQueue>,
+        requestedAssetType: RequestedAssetType,
+    ) {
+        const requestQueue = _.get(requestQueueByNetworkId, req.params.networkId);
+        if (_.isUndefined(requestQueue)) {
+            res.status(400).send('UNSUPPORTED_NETWORK_ID');
+            return;
+        }
+        const didAddToQueue = requestQueue.add(req.params.recipient);
+        if (!didAddToQueue) {
+            res.status(503).send('QUEUE_IS_FULL');
+            return;
+        }
+        utils.consoleLog(
+            `Added ${req.params.recipient} to queue: ${requestedAssetType} networkId: ${req.params.networkId}`,
+        );
+        res.status(200).end();
+    }
+    private static _createProviderEngine(rpcUrl: string) {
+        const engine = new ProviderEngine();
+        engine.addProvider(new NonceSubprovider());
+        engine.addProvider(new HookedWalletSubprovider(idManagement));
+        engine.addProvider(
+            new RpcSubprovider({
+                rpcUrl,
+            }),
+        );
+        engine.start();
+        return engine;
+    }
     constructor() {
         _.forIn(rpcUrls, (rpcUrl: string, networkId: string) => {
-            const providerObj = this._createProviderEngine(rpcUrl);
+            const providerObj = Handler._createProviderEngine(rpcUrl);
             const web3 = new Web3(providerObj);
+            const zeroExConfig = {
+                networkId: +networkId,
```

Comment:
Are these networkIds numbers now in rpc_urls? The +networkId threw me for a bit.

Reply:
yea although they are written as numbers in the literal representation they are converted to string under the hood because JavaScript, I'll change the literal representation back to strings so its not confusing